ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/range/RangeHeaders.java

```diff
@@ -14,29 +14,21 @@
  */
 package org.hyperledger.besu.ethereum.eth.sync.range;
 
-import static com.google.common.base.Preconditions.checkArgument;
-
 import org.hyperledger.besu.ethereum.core.BlockHeader;
 
 import java.util.List;
 import java.util.Objects;
+import java.util.Optional;
 
 import com.google.common.base.MoreObjects;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 public class RangeHeaders {
-  private static final Logger LOG = LoggerFactory.getLogger(RangeHeaders.class);
 
   private final SyncTargetRange range;
   private final List<BlockHeader> headersToImport;
 
   public RangeHeaders(
       final SyncTargetRange checkpointRange, final List<BlockHeader> headersToImport) {
-    if (headersToImport.isEmpty()) {
-      LOG.debug(String.format("Headers list empty. Range: %s", checkpointRange.toString()));
-    }
-    checkArgument(!headersToImport.isEmpty(), "Must have at least one header to import");
```

Comment:
Only question here is if this changes the sync process in any meaningful way - could this cause sync to hang, for example, if we have a peer that is repeatedly returning nothing?  It seems that previously we would throw here, and I'm guessing the sync process would be interrupted because of the exception.

Reply:
@ajsutton - thoughts?

Reply:
Throwing the exception did not disconnect the peer, so we still had the peer.  

But per spec (a) an empty response was 100% fine and (b) Erigon's response was not empty, we just trimmed off the one value we already had.  So there are two routes we may be passing in an empty resposne down the pipeline and both cases the peer is behaving validly, so disconnecting doesn't seem right.

Reply:
Not saying we should disconnect, just want to make sure we're not introducing anything that could cause the sync to stall / get stuck.  But we handle useless responses [elsewhere](https://github.com/hyperledger/besu/blob/f885c4671333168089a7b86ffedf985fce81954b/ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthPeer.java#L215), so even if we got stuck asking for the same range of headers from a single unresponsive peer, I think we would eventually disconnect and move on ...

Reply:
I haven't seen any pauses.  Rather than breaking the pipe an empty stream is the result, and that would get aggregated the same as a broken step, as nothing gets added to the next set of object for the next step in the pipeline.  I haven't seen any slowdown overnight and I started a fresh sync and it's proceeding the same as before with no observable differences.

Side note: I'm really looking forward to being able to bit-torrent the blocks prior to the merge.  EIP-4444 and friends will be welcomed with open arms.